### PR TITLE
Speed up lwt cql tests and also dtest.py refactorings

### PR DESCRIPTION
**Pull request by ptnapoleon**
@smccarthy788 and @mambocab to review. Commit history is good. Running CI now. The PR itself isn't very atomic, but that's okay? The dtest.py changes will allow me to write better code in the follow-up PR, this is mostly defense against needing to rebase

### CODE REVIEW

**Comment by ptnapoleon** [comment]
http://cassci.datastax.com/view/Dev/view/ptnapoleon/job/ptnapoleon-trunk-dtest/lastCompletedBuild/testReport/

**Comment by smccarthy788** [comment]
Any idea how much faster the tests run?

I'm +1

File: dtest.py [hunk 1]
```diff
@@ -764,6 +738,36 @@ def dump_jfr_recording(self, nodes):
             debug(stderr)
 
 
+def get_auth_provider(user, password):
```
Comment (mambocab) [inline dtest.py]: Over the long haul, these functions should get moved to the `tools` module. I'm fine with leaving them here for now, though.

**Comment by ptnapoleon** [comment]
2 minutes

File: dtest.py [hunk 2]
```diff
@@ -737,6 +693,52 @@ def dump_jfr_recording(self, nodes):
             debug(stdout)
             debug(stderr)
 
+# We default to UTF8Type because it's simpler to use in tests
+def create_cf(session, name, key_type="varchar", speculative_retry=None, read_repair=None, compression=None,
```
Comment (mambocab) [inline dtest.py]: Same comment as elsewhere -- go ahead and get this PR in, but could you file a ticket to get these bad boys into `tools`?

**Comment by ptnapoleon** [comment]
These were my guinea pigs for moving something more complicated than thrift tests to reusable cluster. Honestly it seems really easy

**Comment by mambocab** [comment]
Fair enough, merge away :+1:

**Pull request merged**